samples/boards/nrf52/mesh/onoff_level_lighting_vnd_app/src/mesh/device_composition.c

```diff
@@ -2335,18 +1702,18 @@ void light_ctl_temp_publish(struct bt_mesh_model *model)
 	}
 
 	bt_mesh_model_msg_init(msg, BT_MESH_MODEL_LIGHT_CTL_TEMP_STATUS);
-	net_buf_simple_add_le16(msg, state->temp);
-	net_buf_simple_add_le16(msg, state->delta_uv);
+	net_buf_simple_add_le16(msg, (u16_t) get_current(CTL_TEMP));
+	net_buf_simple_add_le16(msg, (s16_t) get_current(CTL_DELTA_UV));
 
-	if (state->temp == state->target_temp &&
-	    state->delta_uv == state->target_delta_uv) {
+	if (state->temp->current == state->temp->target &&
+	    state->duv->current == state->duv->target) {
```

Comment:
Yes, in case of all _publish it is redundant to compare current & target value.
I will replace it by

if (state->transition->counter == 0) {
    goto publish;
} else {
}

But same we can't do in case of all _get()

Reply:
Yes. it is redundant. We should call publish only when target /= current. In that case, it is not necessary to compare them. I will remove it.

Reply:
done !!